api/features/views.py

```diff
@@ -182,18 +182,33 @@ def get_queryset(self):  # type: ignore[no-untyped-def]
             page = self.paginate_queryset(queryset)
             self.environment = Environment.objects.get(id=environment_id)
             self.feature_ids = [feature.id for feature in page]
-            q = Q(
+            feature_states_query = Q(
                 feature_id__in=self.feature_ids,
                 identity__isnull=True,
                 feature_segment__isnull=True,
             )
             feature_states = get_environment_flags_list(
                 environment=self.environment,
-                additional_filters=q,
+                additional_filters=feature_states_query,
                 additional_select_related_args=["feature_state_value", "feature"],
             )
             self._feature_states = {fs.feature_id: fs for fs in feature_states}
 
+            if segment_id := query_data.get("segment"):
```

Comment:
This API is already doing way too much. Have we thought about maybe moving this to a new endpoint?

Reply:
Yes, (partially) agreeing with you. I was thinking the same but in a way we need this segment logic to be related to the feature listing one to share the pagination. That's why the first version couldn't work for customers with > 50 features.
To give some more context. This parameter is only used in the segment page, and we control it. So this does not impact at all . We could definitely separate into another endpoint but it would repeat a big chunk of this logic.
Wdyt ?

Reply:
Ugh, true. I don't think I have any other solution apart from a major refactor, but that's for another day.